src/app/shared/components/carousel/carousel.component.ts

```diff
@@ -0,0 +1,40 @@
+import { Component, OnInit, AfterContentInit } from '@angular/core';
+import { ViewChild, ElementRef } from '@angular/core';
+
+@Component({
+  selector: 'app-carousel',
+  templateUrl: './carousel.component.html',
+  styleUrls: ['./carousel.component.scss']
+})
+export class CarouselComponent implements OnInit, AfterContentInit {
+  private currTranslate = 0;
+  private childrenLength: number;
+
+  @ViewChild('slidesContainer') slidesContainer: ElementRef;
+
+  ngOnInit(): void {
+    this.slidesContainer.nativeElement.style.transform = `translate(${0}%)`;
+  }
+
+  ngAfterContentInit(): void {
+    this.childrenLength = this.slidesContainer.nativeElement.children.length;
+  }
+
+  public nextSlide(): void {
+    const sliderLengthCheck: number = (-this.childrenLength / 4 + 1) * 100;
```

Comment:
the variable name is not clear. The type is number but name is check?

Also, move 4 in a constant variable outside the class and give the representation name

like const ITEM_PER_PAGE = 4

![wat8](https://user-images.githubusercontent.com/9499109/51175566-59746380-18c3-11e9-80f4-681858b9c690.jpg)